sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -124,67 +124,122 @@ case class GpuElementAt(left: Expression, right: Expression, failOnError: Boolea
   // GetArrayItemUtil.computeNullabilityFromArray
   override def nullable: Boolean = true
 
+  @transient
+  private lazy val doElementAtV: (ColumnView, ColumnView) => cudf.ColumnVector =
+    left.dataType match {
+      case _: ArrayType =>
+        (array, indices) => {
+          if (failOnError) {
+            // Check if any index is out of bound only when ansi mode is enabled. Nulls in either
+            // array or indices are skipped during this computation. Then no exception will be
+            // raised if no valid entry (An entry is valid when both the array row and its index
+            // are not null), the same with what Spark does.
+            val hasLargerIndices = withResource(indices.abs()) { absIndices =>
+              withResource(array.countElements()) { numElements =>
+                absIndices.greaterThan(numElements)
+              }
+            }
+            withResource(hasLargerIndices) { _ =>
+              if(BoolUtils.isAnyValidTrue(hasLargerIndices)) {
+                throw new ArrayIndexOutOfBoundsException("Some indices are out of bound")
```

Comment:
We are not using the RapidsErrorUtils. Is it worth finding the first entry that does not match after this point and getting the exact same error message?

Reply:
I think it would be OK since we are on the way to throw an exception to abort.
The main drawback is GPU memory, we need to hold the 'numElements' column until the whole check is done.